python/ml4ir/applications/ranking/tests/test_ranklib_in_ml4ir.py

```diff
@@ -0,0 +1,125 @@
+
+INPUT_DIR = "ml4ir/applications/ranking/tests/data/ranklib_test_data/"
+QUERY_ID_NAME = 'qid'
+RELEVANCE_NAME = 'relevance'
+KEEP_ADDITIONAL_INFO = 0
+NON_ZERO_FEATURES_ONLY = 0
+
+
+class TestRanklibConversion(unittest.TestCase):
+    def setUp(self):
+        self.feature_config_yaml = INPUT_DIR + 'feature_config.yaml'
+        self.feature_config_yaml_convert_to_clicks = INPUT_DIR + 'feature_config_convert_to_clicks.yaml'
+
+    def parse_config(self, tfrecord_type: str, feature_config, io) -> SequenceExampleFeatureConfig:
+        if feature_config.endswith(".yaml"):
+            feature_config = io.read_yaml(feature_config)
+        else:
+            feature_config = yaml.safe_load(feature_config)
+
+        return SequenceExampleFeatureConfig(feature_config, None)
+
+    def test_ranklib_in_ml4ir(self):
+        io = local_io.LocalIO()
+        exFeatureConfig = self.parse_config(TFRecordTypeKey.SEQUENCE_EXAMPLE, self.feature_config_yaml, io)
+        preprocessing_keys_to_fns = {}
+        if 'preprocessing_info' in exFeatureConfig.get_label():
+            if exFeatureConfig.get_label()['preprocessing_info'][0]['fn'] == 'convert_label_to_clicks':
+                preprocessing_keys_to_fns['convert_label_to_clicks'] = convert_label_to_clicks
+
+        dataset = RelevanceDataset(
+            data_dir=INPUT_DIR,
+            data_format=DataFormatKey.RANKLIB,
+            feature_config=exFeatureConfig,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            batch_size=1,
+            file_io=io,
+            preprocessing_keys_to_fns=preprocessing_keys_to_fns,
+            logger=None,
+            keep_additional_info=KEEP_ADDITIONAL_INFO,
+            non_zero_features_only=NON_ZERO_FEATURES_ONLY,
+            max_sequence_size=319,
+        )
+        non_one_hot = False
+        chk = [e for e in dataset.train]
+        for e in chk:
+            if sum(e[1][0]).numpy() > 1:
+                non_one_hot = True
+                break
+        assert non_one_hot == True
+        assert len(chk) == 49
+
+        non_one_hot = False
+        chk = [e for e in dataset.validation]
+        for e in chk:
+            if sum(e[1][0]).numpy() > 1:
+                non_one_hot = True
+                break
+        assert non_one_hot == True
+        assert len(chk) == 49
+
+        non_one_hot = False
+        chk = [e for e in dataset.test]
+        for e in chk:
+            if sum(e[1][0]).numpy() > 1:
+                non_one_hot = True
+                break
+        assert non_one_hot == True
+        assert len(chk) == 49
+
+        exFeatureConfig = self.parse_config(TFRecordTypeKey.SEQUENCE_EXAMPLE, self.feature_config_yaml_convert_to_clicks, io)
+        preprocessing_keys_to_fns = {}
+        if exFeatureConfig.get_label()['preprocessing_info'][0]['fn'] == 'convert_label_to_clicks':
+            preprocessing_keys_to_fns['convert_label_to_clicks'] = convert_label_to_clicks
+
+        dataset = RelevanceDataset(
```

Comment:
@mohazahran Break it up into smaller tests. I understand what you are trying to do, but will be better if you break it up IMO.